ssl/statem/statem.c

```diff
@@ -157,13 +157,8 @@ int ossl_statem_skip_early_data(SSL *s)
     if (s->ext.early_data != SSL_EARLY_DATA_REJECTED)
         return 0;
 
-    if (s->hello_retry_request) {
-        if (s->statem.hand_state != TLS_ST_SW_HELLO_RETRY_REQUEST)
```

Comment:
Consider the case

ClientHello---------------->
stream EarlyData-------->
                 <-------------------HelloRetryRequest
still have some EarlyData in flight------->
ClientHello2--------------->
                 <-------------------ServerHello

Since the network is not instantaneous, we still need to be able to skip over early data records until we receive another handshake record.  (See the list of three behaviors in section 4.2.9 of draft-21.)